nifi-toolkit/nifi-toolkit-tls/src/test/java/org/apache/nifi/toolkit/tls/util/TlsHelperTest.java

```diff
@@ -448,4 +460,58 @@ public void testClientDnFilenameSpecialChars() throws Exception {
         assertEquals("CN=testuser_OU=NiFi_Organisation", escapedClientDn);
     }
 
+    private KeyStore setupKeystore() throws Exception {
+
+        KeyStore ks = KeyStore.getInstance("JKS");
+        InputStream readStream = getClass().getClassLoader().getResourceAsStream("keystore.jks");
+        ks.load(readStream, "changeit".toCharArray());
+
+        return ks;
+    }
+
+    @Test
+    public void testOutputCertsAsPem() throws Exception {
+        File folder = tempFolder.newFolder("splitKeystoreOutputDir");
+        KeyStore keyStore = setupKeystore();
+        HashMap<String, Certificate> certs = TlsHelper.extractCerts(keyStore);
+        TlsHelper.outputCertsAsPem(certs, folder,".crt");
+
+        for(File file : folder.listFiles()) {
+            BufferedReader br = new BufferedReader(new FileReader(file));
+            PEMParser pemParser = new PEMParser(br);
+            X509CertificateHolder key = (X509CertificateHolder) pemParser.readObject();
+            assertNotNull(key.getSignature());
+        }
+    }
+
+    @Test
+    public void testOutputKeysAsPem() throws Exception {
+        File folder = tempFolder.newFolder("splitKeystoreOutputDir");
+        KeyStore keyStore = setupKeystore();
+        HashMap<String, Key> keys = TlsHelper.extractKeys(keyStore, "changeit".toCharArray());
+        TlsHelper.outputKeysAsPem(keys, folder, ".key");
+
+        for(File file : folder.listFiles()) {
+            BufferedReader br = new BufferedReader(new FileReader(file));
+            PEMParser pemParser = new PEMParser(br);
+            PEMKeyPair key = (PEMKeyPair) pemParser.readObject();
+            assertNotNull(key.getPrivateKeyInfo());
+        }
+    }
+
+    @Test
+    public void testExtractCerts() throws Exception {
+        KeyStore keyStore = setupKeystore();
+        HashMap<String, Certificate> certs = TlsHelper.extractCerts(keyStore);
+        assertEquals(2, certs.size());
+        certs.forEach((String p, Certificate q) -> assertEquals("X.509", q.getType()));
+    }
+
+    @Test
+    public void testExtractKeys() throws Exception {
+        KeyStore keyStore = setupKeystore();
+        HashMap<String, Key> keys = TlsHelper.extractKeys(keyStore, "changeit".toCharArray());
+        assertEquals(1, keys.size());
+        keys.forEach((String alias, Key key) -> assertEquals("PKCS#8", key.getFormat()));
+    }
```

Comment:
I would like to see test cases documenting the following use cases: 

* Run with no keystore provided
* Run with no keystore password provided
* Run with wrong keystore password provided
* Run with keystore with no aliases
* Run with keystore with multiple aliases
* Run with correct keystore password but wrong key password provided